packages/react-reconciler/src/ReactFiberBeginWork.js

```diff
@@ -1923,6 +1926,70 @@ function updateContextConsumer(
   return workInProgress.child;
 }
 
+function updateEventComponent(current, workInProgress, renderExpirationTime) {
+  const nextProps = workInProgress.pendingProps;
+  let nextChildren = nextProps.children;
+
+  reconcileChildren(
+    current,
+    workInProgress,
+    nextChildren,
+    renderExpirationTime,
+  );
+  if (__DEV__) {
+    // Check event node has any text nodes as direct nodes and warn
+    let child = workInProgress.child;
+    while (child !== null) {
+      if (child.tag === HostText) {
+        warningWithoutStack(
```

Comment:
This won’t warn if the node is nested in a function component for example. We could use the host context for this instead. We don’t have to do that in this PR but can you add a test for the warning and maybe a commented out test for the nested case? That way it’s easy to do that change later.

Reply:
Why exclude the stack here? Seems useful to have the stack in the warning.